### DISCUSSION TITLE
Checking a seamless image

### DISCUSSION THREAD

**Discussion by keto33** [ImageMagick/ImageMagick#3188] [Help]
I look for a command to check if an image is seamless or not.

In a seamless image, the pixels of left and right columns/top and bottom rows should have matching colours. Thus, when repeating/tiling them, they show no mismatch. For example, by

`magick -size 800x600 tile:pattern.jpg large.jpg
`
we can see the first image is seamless (a large image is created as the small image therein is not recognisable), but the second image is not seamless (we see the boundaries of repetition).

![d](https://user-images.githubusercontent.com/66027046/106660019-69893980-6597-11eb-90fd-c13191deb121.jpg)
![c](https://user-images.githubusercontent.com/66027046/106660045-727a0b00-6597-11eb-80a9-7c4076c8aa52.jpg)

**Comment by keto33** [comment]
With the above idea, I came with

```
magick img.jpg \( -clone 0 -gravity East -crop 1x+0+0 \) \( -clone 0 -gravity West -crop 1x+0+0 \) -compose Difference -composite -colorspace gray -format '%[fx:mean*100]' info:

magick img.jpg \( -clone 0 -gravity North -crop x1+0+0 \) \( -clone 0 -gravity South -crop x1+0+0 \) -compose Difference -composite -colorspace gray -format '%[fx:mean*100]' info:
```

but it does not work as expected (or probably wrong expectionation)

**Comment by snibgo** [comment]
I have looked at this problem, and I can't find a solution. For example, we can append two instances of your second input side by side, and compare each pair of columns. Does the central pair have a statistic that marks it out from all other pairs? I can't find any such statistic.
